.travis.yml

```diff
@@ -42,61 +42,50 @@ matrix:
 
-        # Try all python versions with the latest numpy, but without the
-        # optional dependencies
-        - os: linux
-          env: PYTHON_VERSION=2.6 SETUP_CMD='test --open-files'
-        - os: linux
-          env: PYTHON_VERSION=2.7 SETUP_CMD='test --open-files'
-        - os: linux
-          env: PYTHON_VERSION=3.3 SETUP_CMD='test --open-files'
-        - os: linux
-          env: PYTHON_VERSION=3.4 SETUP_CMD='test --open-files'
-        - os: linux
-          env: PYTHON_VERSION=3.5 SETUP_CMD='test --open-files'
-
-        # Now try with all optional dependencies on 2.7 and an appropriate 3.x
-        # build (with latest numpy). We also note the code coverage on Python
-        # 2.7.
+        # Try the full monty, with all optional dependencies on an appropriate
+        # 3.x build(noting the coverage) and on 2.7, both with latest numpy.
         - os: linux
-          env: PYTHON_VERSION=2.7 SETUP_CMD='test --coverage'
+          env: SETUP_CMD='test --coverage'
                CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
                PIP_DEPENDENCIES='cpp-coveralls'
                LC_CTYPE=C.ascii LC_ALL=C.ascii
                CFLAGS='-ftest-coverage -fprofile-arcs -fno-inline-functions -O0'
         - os: linux
-          env: PYTHON_VERSION=3.4 SETUP_CMD='test'
+          env: PYTHON_VERSION=2.7 SETUP_CMD='test'
                CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
                LC_CTYPE=C.ascii LC_ALL=C.ascii
 
-        # Try older numpy versions without optional dependencies
-        - os: linux
-          env: PYTHON_VERSION=2.7 NUMPY_VERSION=1.8 SETUP_CMD='test'
+        # Try other python versions with the latest numpy but without the
+        # optional dependencies
         - os: linux
-          env: PYTHON_VERSION=2.7 NUMPY_VERSION=1.7 SETUP_CMD='test'
+          env: PYTHON_VERSION=3.3 SETUP_CMD='test --open-files'
         - os: linux
-          env: PYTHON_VERSION=2.7 NUMPY_VERSION=1.6 SETUP_CMD='test'
+          env: PYTHON_VERSION=3.4 SETUP_CMD='test --open-files'
 
-        # Try developer version of Numpy without optional dependencies
+        # Try older numpy versions on python 2.7, without optional dependencies
         - os: linux
-          env: PYTHON_VERSION=2.7 NUMPY_VERSION=dev SETUP_CMD='test'
+          env: PYTHON_VERSION=2.7 NUMPY_VERSION=1.8 SETUP_CMD='test'
+        - os: linux
+          env: PYTHON_VERSION=2.7 NUMPY_VERSION=1.7 SETUP_CMD='test'
 
         # Do a PEP8 test
         - os: linux
-          env: PYTHON_VERSION=2.7 MAIN_CMD='pep8 astropy --count' SETUP_CMD=''
+          env: MAIN_CMD='pep8 astropy --count' SETUP_CMD=''
 
     allow_failures:
-      - env: PYTHON_VERSION=2.7 NUMPY_VERSION=dev SETUP_CMD='test'
+        # test with numpy development; we want to know when there are problems,
+        # but not necessarily be unable to move forward ourselves, so allow this
+        # to fail.
+        - env: NUMPY_VERSION=dev SETUP_CMD='test'
```

Comment:
you need to list this above in the matrix too, and list the duplicate of the exact same env call under ``allow_failures``.